Content.Shared/Verbs/Verb.cs

```diff
@@ -1,90 +1,143 @@
-﻿using System;
-using JetBrains.Annotations;
 using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using System;
 
 namespace Content.Shared.Verbs
 {
+    [Flags] public enum VerbType
+    {
+        Interaction = 1,
+        Activation = 2,
+        Alternative = 4,
+        Other = 8,
+        All = 1+2+4+8
+    }
+
     /// <summary>
-    ///     A verb is an action in the right click menu of an entity.
+    ///     Verb objects describe actions that a user can take. The actions can be specified via an Action, local
+    ///     events, or networked events. Verbs also provide text, icons, and categories for displaying in the
+    ///     context-menu.
     /// </summary>
-    /// <remarks>
-    ///     To add a verb to an entity, define it as a nested class inside the owning component,
-    ///     and mark it with <see cref="VerbAttribute"/>
-    /// </remarks>
-    [UsedImplicitly]
-    public abstract class Verb : VerbBase
+    [Serializable, NetSerializable]
```

Comment:
why is this serializable/netserializable? i dont think this can be sent over the wire (esp. since delegates and events cant be serialized?)

Reply:
Previously there was a dedicated serializable `NetVerbData` struct that had fields for the verb text, key, category, icon, and visibility. This was used to add server-side verbs to the the client's context menu.

At some point I realised that Verb and NetverbData were basically identical except for the delegate & events. So I just made Verb serializable and marked the delegate & events with the [NonSerialized] attribute.

As far as I'm concerned, the netcode may as well be black magic. So maybe I wasn't meant to do that but it seems to work and just doesn't send those nullable fields to the client.